hugr-core/src/import.rs

```diff
@@ -865,6 +805,8 @@ impl<'a> Context<'a> {
             self.record_links(exit, Direction::Incoming, region_data.targets);
         }
 
+        self.region_scope = prev_region;
```

Comment:
For this it would be possible. However it wouldn't be possible for stuff like `SymbolTable` due to borrow checker limitations. Using `enter` and `exit` for `SymbolTable` and the other scoping utilities is also not entirely ideal (a Python-like `with` would be cool), but it showed me that perhaps complexity-wise avoiding the closure isn't a terrible tradeoff.

Reply:
finite number of places to forget to exit I guess